### ISSUE TITLE
next-unread-mailbox doesn't jump to a mailbox with unread messages

### ISSUE THREAD

**Issue by lasombra** [neomutt/neomutt#16]
Instead it present this error message: "No mailboxes have new mail". Bind is:

`bind index v next-unread-mailbox`

Same thing happens if I add pages to the bind.

**Comment by flatcap** [comment]
@lasombra 
I've just tried NeoMutt, Mutt-1.6.0 and Mutt-1.5.24 and none of them work for me.
When did this last work correctly?
Did it **ever** work for you?

**Comment by flatcap** [comment]
@lasombra 
The Sidebar functions `<sidebar-next-new>` and `<sidebar-prev-new>` work fine.
I'll keep looking into what's happening with `<next-unread-mailbox>`.

**Comment by lasombra** [comment]
@flatcap If it ever worked? Good question. It's been a while since I last tried mutt but I remember it working in the past, so much so that it is on my old configuration file, but if `<sidebar-next-new>` and `<sidebar-prev-new>` work fine, then it's no problem at all, just a matter of start using it :)

Thanks for looking into it.

**Comment by flatcap** [comment]
Is your mail local, in maildir directories?
If so, have you set `$maildir_check_cur`?
By default, Mutt only looks in the `new` directory (ignoring old-but-unread emails in the `cur` directory).

**Comment by flatcap** [comment]
No problem.

**Comment by lasombra** [comment]
It's in Maildir directories and I have `$check_new` set only. I'll give it a try tomorrow and will let you know how it goes.

**Comment by lasombra** [comment]
So, `macro index v '<sidebar-next-new><sidebar-open>'` and `macro index V '<sidebar-prev-new><sidebar-open>'` work well, except that it also considers mailboxes with `[number]` as mailboxes with new messages. I thought `[number]` represented the number of flagged/important messages on a folder.

Also, `$maildir_check_cur` worked only once for me when using `<next-unread-mailbox>`.

**Comment by flatcap** [comment]
> `<sidebar-next-new>` and `<sidebar-prev-new>` work well

good

> except that it also considers mailboxes with `[number]` as mailboxes with new messages.
> I thought `[number]` represented the number of flagged/important messages on a folder.

Ah, you got me.  That _is_ inconsistent (and undocumented).

I've changed the Sidebar, to do what you expect.

> `$maildir_check_cur` worked only once for me when using `<next-unread-mailbox>`.

There's something not right with this.
I'll keep looking.

**Comment by lasombra** [comment]
@flatcap Brilliant, thanks for the help!

**Comment by simmel** [comment]
`next-unread-mailbox` has worked for me since I started using mutt and at least 1.5.23. This has both worked for me with local Maildir (synced via offlineimap) and IMAP.

`set mark_old=no` is the only relevant setting I can think of that I've set. I'm happy to provide whole or parts of my `.muttrc`.

Thanks for the workaround via `<sidebar-next-new>` and `<sidebar-prev-new>`!

**Comment by lasombra** [comment]
@simmel Thanks for the tip, still only works for the first mailbox jump, not subsequent ones :(

This tells me that the binding and functionality work but something happens after the first mailbox jump.

**Comment by simmel** [comment]
On a somewhat related note; it's pretty annoying that `<sidebar-next-new><sidebar-open>` needs the sidebar to open to work.

**Comment by lasombra** [comment]
Well... at least you can solve it with a bind. I think the original sidebar
patch worked the same way IIRC.

**Comment by simmel** [comment]
On Wed, 2016-04-27 at 01:03:00 -0700, LaSombra wrote:
> Well... at least you can solve it with a bind.

Ah, of course! Thank you!
```
macro index,pager <Tab> '<enter-command>set sidebar_visible=yes<enter><sidebar-next-new><sidebar-open>'
```
worked perfectly!

> I think the original sidebar patch worked the same way IIRC.

Yeah probably, I've just never used <sidebar-next-new>.

Let us know when we can test your patch.

**Comment by flatcap** [comment]
@simmel 
> it's pretty annoying that `<sidebar-next-new><sidebar-open>` needs the sidebar to open to work.

Hmm... That's a result of my cautious programming.
However, you're right, there's no good reason for the `<sidebar-*>` functions not to work.

I've fixed the "problem" in 6e2a09b which will be in the next NeoMutt release.

**Comment by simmel** [comment]
On a related note @flatcap, mutt doesn't update that it has any new mail in any folders if the sidebar is closed.

**Comment by flatcap** [comment]
@simmel 
I think that has to go in the "won't-fix" category (for now).  If the user hasn't enabled the sidebar, I don't want to be doing lots of disk/network-heavy activity.

However, I'll add the idea to my notes for the "refresh" overhaul.  A middle-ground `sidebar_refresh_when_hidden` option might be useful.  **Would you like to implement it?**

**P.S.**  If you have any more good ideas / bugs (and I really welcome them), please open a new issue for each.  That makes it easier for everyone to follow.

**Comment by simmel** [comment]
On Fri, 2016-04-29 at 03:27:15 -0700, Richard Russon wrote:
> However, I'll add the idea to my notes for the "refresh" overhaul.  A middle-ground `sidebar_refresh_when_hidden` option might be useful.  **Would you like to implement it?**

I would but I won't have time for it in the forseeable future I'm
afraid.

> **P.S.**  If you have any more good ideas / bugs (and I really welcome them), please open a new issue for each.  That makes it easier for everyone to follow.

Of course!

**Comment by protist** [comment]
This workaround is fine when you are already in mutt. However, I launch mutt with `mutt -Z`, and this is broken for neomutt. Normally, from the `man` page, this "causes Mutt to open the first mailbox specified by the mailboxes command which contains new mail." This works in vanilla mutt 1.6.0 (as does `<next-unread-mailbox>`), but not in neomutt, which just opens the default mailbox instead.

**Comment by flatcap** [comment]
@protist 
OK, I see the problem.  I'll see what I can do.

**Comment by protist** [comment]
Thank you for that, @flatcap.

There's another reason for addressing this directly. `<next-unread-mailbox>` jumps to any mailbox that contains new mail. OTOH, `<sidebar-next-new>` will only move to unread mail if it's in a mailbox in the sidebar that is after the current mailbox. For example, if there is unread mail in a mailbox in the sidebar that *above* the current mailbox, it won't work (i.e. you'd have to bind another key).

**Comment by flatcap** [comment]
@protist 
It seems that the sidebar refresh limit is getting in the way of some normal operations.
I'm not suggesting this as a solution to your problem, but...
If you `set sidebar_refresh_time=1` then `<next-unread-mailbox> ` starts working correctly.

Note: `<sidebar-next-new>` (and prev) are modified by `$sidebar_next_new_wrap`.

Still working...

**Comment by protist** [comment]
Thanks for the update @flatcap. This sounds like some good progress in isolating the problem. You might have already tested, but just in case… `set sidebar_refresh_time=1` works as you say, but doesn't specifically fix the `mutt -Z` problem. Thanks for all the information, though.

**Comment by lasombra** [comment]
Good morning, it seems this issue is back on the latest git version of branch `neomutt`.

**Comment by flatcap** [comment]
@lasombra 
Thanks.  I'm sorry I haven't done much about this issue.  The Mutt devs are making large changes underneath me and I don't want to make changes that will be immediately obsolete.  That said, I've been a bit lazy too.  I shall make time for this.

**Comment by lasombra** [comment]
@flatcap No worries man. I appreciate you taking the time to look at it and I know fast moving targets are hard to track. You are doing some fine work and I thank you for this.

**Comment by protist** [comment]
Oh! `mutt -Z` now seems to work for me. I last updated to `NeoMutt 2016-06-11` on 13 Jun, but I've only just noticed that it's working fine now. I'm not sure if that update fixed it, or I changed something in my muttrc. Anyway, it's all good now. Cheers!

**Comment by protist** [comment]
I tested commit 38f2bb0, and `mutt -Z` was broken again. However, I tried the latest commit 0b41c85, and it's working again! I'm not really sure why, but feel free to close this issue.
